src/main/java/com/powsybl/openloadflow/network/impl/LfLoadImpl.java

```diff
@@ -30,15 +30,15 @@ public class LfLoadImpl extends AbstractLfInjection implements LfLoad {
 
     private final LfLoadModel loadModel;
 
-    private final Map<String, Ref<Load>> loadsRefs = new LinkedHashMap<>();
+    private final Map<String, Ref<Load>> loadsRefs = new HashMap<>();
 
     private final List<Ref<LccConverterStation>> lccCsRefs = new ArrayList<>();
 
     private double targetQ = 0;
 
     private boolean ensurePowerFactorConstantByLoad = false;
 
-    private final List<Double> loadsAbsVariableTargetP = new ArrayList<>();
+    private final HashMap<String, Double> loadsAbsVariableTargetP = new HashMap<>();
```

Comment:
Do we need to define the object as the implementation class? @vidaldid-rte 

`Map<String, Double>` instead of `HashMap<String, Double>`

Reply:
You lean define it as a Map ? It avoids some pedantic discussions with Sonar lovers so why not, although it does not really matter in a private field definition.
There is still an issue with that - in theory it slows down iterations on the array (that happen frequently) because whe would break the processor cache here. I don't know how much it matters in real perf.
But we may revert this change and instead add commentarys that it is important to update the array and the map in sync because the code relies on the order.